tools/ci_connector_ops/ci_connector_ops/pipelines/pipelines/metadata.py

```diff
@@ -26,6 +28,64 @@ async def _run(self) -> StepStatus:
         return await self.get_step_result(run_test)
 
 
+class SimpleExecStep(Step):
+    def __init__(self, context: PipelineContext, title: str, args: List[str]):
+        self.title = title
+        self.args = args
+        super().__init__(context)
+
+    async def _run(self) -> StepStatus:
+        run_test = self.context.dagger_client.with_exec(self.args)
+        return await self.get_step_result(run_test)
+
+
+def metadata_validation_step(metadata_pipeline_context: PipelineContext, metadata_path: Path) -> Step:
+    return SimpleExecStep(
+        context=metadata_pipeline_context,
+        title=f"Validate Connector Metadata Manifest: {metadata_path}",
+        args=["metadata_service", "validate", str(metadata_path)],
+    )
+
+
+async def run_metadata_validation_pipeline(
+    is_local: bool,
+    git_branch: str,
+    git_revision: str,
+    gha_workflow_run_url: Optional[str],
+    pipeline_start_timestamp: Optional[int],
+    ci_context: Optional[str],
+    metadata_source_paths: Set[Path],
+) -> bool:
+    metadata_pipeline_context = PipelineContext(
+        pipeline_name="Metadata Service Validation Pipeline",
+        is_local=is_local,
+        git_branch=git_branch,
+        git_revision=git_revision,
+        gha_workflow_run_url=gha_workflow_run_url,
+        pipeline_start_timestamp=pipeline_start_timestamp,
+        ci_context=ci_context,
+    )
+
+    async with dagger.Connection(DAGGER_CONFIG) as dagger_client:
+        # TODO refactor the environemnts to use containers not contexts
+        metadata_pipeline_context.dagger_client = dagger_client.pipeline(metadata_pipeline_context.pipeline_name)
+        updated_client = with_pipx_module(
+            metadata_pipeline_context, ".", f"{METADATA_DIR}/{METADATA_LIB_MODULE_PATH}", include=["airbyte-integrations/connectors/*"]
+        )
+
+        metadata_pipeline_context.dagger_client = updated_client
```

Comment:
Why are you reassigning the client here?

Reply:
As discussed in our last call I don't think the re-assignation should happen here.